### PR TITLE
Enable changing header visibility in TopUpWithFiat component

### PR DESCRIPTION
**Pull request by JustynaBroniszewska**

### CODE REVIEW

File: universal-login-react/src/ui/TopUp/Fiat/TopUpWithFiat.tsx [hunk 1]
```diff
@@ -51,6 +52,7 @@ export const TopUpWithFiat = ({walletService, modalClassName, logoColor}: TopUpW
         </>
       );
     case TopUpProvider.RAMP:
+      setHeaderVisible(false);
```
Comment (Marik-D) [inline universal-login-react/src/ui/TopUp/Fiat/TopUpWithFiat.tsx]: Side effects in render are bad practice

Comment (Marik-D) [reply universal-login-react/src/ui/TopUp/Fiat/TopUpWithFiat.tsx]: Instead do `useEffect` on `modal` and call it there

File: universal-login-react/src/ui/TopUp/Fiat/TopUpWithFiat.tsx [hunk 2]
```diff
@@ -15,10 +15,11 @@ export interface TopUpWithFiatProps {
   walletService: WalletService;
   logoColor?: LogoColor;
   modalClassName?: string;
+  setHeaderVisible: (arg: boolean) => void;
```
Comment (Marik-D) [inline universal-login-react/src/ui/TopUp/Fiat/TopUpWithFiat.tsx]:
```suggestion
  setHeaderVisible: (isVisible: boolean) => void;
```

File: universal-login-react/src/ui/TopUp/Fiat/TopUpWithFiat.tsx (hunk 1 above)
Comment (Marik-D) [reply universal-login-react/src/ui/TopUp/Fiat/TopUpWithFiat.tsx]: Also not sure if a callback to the parent component is the best solution here. Consider moving the state up and/or decomposing this component into two: `TopUpFiatDetails` and `TopUpWithOnRampProvider`

Comment (JustynaBroniszewska) [reply universal-login-react/src/ui/TopUp/Fiat/TopUpWithFiat.tsx]: Good point with `useEffect` - done

File: universal-login-react/src/ui/TopUp/Fiat/TopUpWithFiat.tsx (hunk 2 above)
Comment (JustynaBroniszewska) [reply universal-login-react/src/ui/TopUp/Fiat/TopUpWithFiat.tsx]: done ✅

File: universal-login-react/src/ui/TopUp/Fiat/TopUpWithFiat.tsx [hunk 3]
```diff
@@ -29,6 +30,14 @@ export const TopUpWithFiat = ({walletService, modalClassName, logoColor}: TopUpW
     setModal(provider);
   };
 
+  useEffect(() => {
+    if (modal === 'none') {
```
Comment (Marik-D) [inline universal-login-react/src/ui/TopUp/Fiat/TopUpWithFiat.tsx]: setHeaderVisble(modal === 'none')

**Comment by Marik-D** [comment]
Or better, just move the `modal` state up, instead of `isHeaderVisible` it will be one less state that way

**Pull request merged**